### PR TITLE
bitcoin: update to 0.21.0.

### PR DESCRIPTION
**Pull request by unspecd**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, x86_64-musl
- [x] I built this PR locally for these architectures:
  - [x] aarch64-musl
  - [x] armv7l
  - [x] armv6l-musl

### CODE REVIEW

File: srcpkgs/bitcoin/template [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
+ libevent-devel qrencode-devel libatomic-devel sqlite-devel"
 short_desc="Bitcoin is a peer-to-peer network based digital currency"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://bitcoin.org/"
 distfiles="https://bitcoin.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz
- https://raw.github.com/bitcoin-core/packaging/master/debian/bitcoin-qt.desktop
- https://raw.github.com/bitcoin/bitcoin/v${version}/share/pixmaps/bitcoin128.png"
-checksum="f2591d555b8e8c2e1bd780e40d53a91e165d8b3c7e0391ae2d24a0c0f23a7cc0
+ https://raw.githubusercontent.com/bitcoin-core/packaging/master/debian/bitcoin-qt.desktop
```
Comment (ericonr) [inline srcpkgs/bitcoin/template:16]: Please don't get it from mater (it's an error from the previous version).

**Review by ericonr — COMMENTED** [review]
Please also mention in the commit message that you are fixing cross so the GUI is always built.

File: srcpkgs/bitcoin/template (hunk 1 above)
Comment (unspecd) [reply srcpkgs/bitcoin/template]: > mater

Do you mean 'master'? Anyway, updated the PR.

File: srcpkgs/bitcoin/template [hunk 2]
```diff
@@ -1,39 +1,49 @@
...
-	subpackages+=" bitcoin-qt"
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" qt5-host-tools"
 fi
 
+pre_configure() {
+	autoreconf -fi
+	case "$XBPS_TARGET_MACHINE" in
+	aarch64*)
+		CFLAGS=${CFLAGS/armv8-a/armv8-a+crc+crypto}
```
Comment (ericonr) [inline srcpkgs/bitcoin/template:32]: Can you explain why this is necessary? Does it fail to build without it?

Comment (unspecd) [reply srcpkgs/bitcoin/template:32]: Yes, it does. Simple test:
```c
/* sample.c */
#include <arm_acle.h>

int main()
{
    __crc32cd(0, 0);
    return 0;
}
```
```console
$ aarch64-linux-musl-gcc sample.c
/usr/lib/gcc/aarch64-linux-musl/10.2/include/arm_acle.h:128:1: error: inlining failed in call to 'always_inline' '__crc32cd': target specific option mismatch
  128 | __crc32cd (uint32_t __a, uint64_t __b)
      | ^~~~~~~~~
sample.c:5:5: note: called from here
    5 |     __crc32cd(0, 0);
```
```console
$ aarch64-linux-musl-gcc -march=armv8-a+crc+crypto sample.c ; echo $?
0
```

**Pull request merged**